core/app/models/spree/order.rb

```diff
@@ -410,7 +410,7 @@ def fulfill!
 
     # Helper methods for checkout steps
     def paid?
-      payments.valid.completed.size == payments.valid.size && payments.valid.sum(:amount) >= total
+      payments.valid.completed.size == payments.valid.size && payments.valid.completed.sum(:amount) >= total
```

Comment:
@MatthewKennedy my only concern here is that some payment gateways complete the purchase in an async way (eg. via webhooks), which would break the current checkout flow

Reply:
@damianlegawiec that should be better.